server.js

```diff
@@ -6892,6 +6892,126 @@ cache((data, match, sendBadge, request) => {
+  const pieces = match.input.split('/');
+  const badgeData = getBadgeData('nsp', data);
+  const nspRequestOptions = {
+    method: 'POST',
+    body: {
+      package: {}
+    },
+    json: true
+  };
+  let format = '';
+
+  function getNspResults (packageVersion = '') {
+    nspRequestOptions.body.package.version = packageVersion;
+
+    request('https://api.nodesecurity.io/check', nspRequestOptions, (error, response, body) => {
+      if (error !== null || typeof body !== 'object' || body === null) {
+        badgeData.text[1] = 'invalid';
+        badgeData.colorscheme = 'red';
+      } else if (body.length !== 0) {
+        badgeData.text[1] = `${body.length} vulnerabilities`;
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = 'no known vulnerabilities';
+        badgeData.colorscheme = 'brightgreen';
+      }
+
+      sendBadge(format, badgeData);
+    });
+  }
+
+  function getNpmVersionThenNspResults (packageName = '') {
+    // nsp doesn't properly detect the package version in POST requests so this function gets it for us
+    // https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#getpackageversion
+    let npmURL;
+    const npmRequestOptions = {
+      headers: {
+        Accept: '*/*'
+      },
+      json: true
+    };
+
+    try {
+      if (packageName[0] === '@') {
+        // Using 'latest' would save bandwidth, but it is currently not supported for scoped packages
+        npmURL = `http://registry.npmjs.org/@${encodeURIComponent(packageName.slice(1))}`;
+      } else {
+        npmURL = `http://registry.npmjs.org/${encodeURIComponent(packageName)}/latest`;
+      }
+    } catch (e) {
+      badgeData.text[1] = 'invalid';
+      badgeData.colorscheme = 'red';
+
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    request(npmURL, npmRequestOptions, (error, response, body) => {
+      if (response !== null && response.statusCode === 404) {
+        // NOTE: in POST requests nsp does not distinguish between 'package not found' and 'no known vulnerabilities'. To keep consistency in the use case where a version is provided (which skips `getNpmVersionThenNspResults()` altogether) we'll say 'no known vulnerabilities' since it is technically true in both cases
+        badgeData.text[1] = 'no known vulnerabilities';
+
+        sendBadge(format, badgeData);
+      } else if (error !== null || typeof body !== 'object' || body === null) {
+        badgeData.text[1] = 'invalid';
+        badgeData.colorscheme = 'red';
+
+        sendBadge(format, badgeData);
+      } else if (typeof body.version === 'string') {
+        getNspResults(body.version);
+      } else if (typeof body['dist-tags'] === 'object') {
+        getNspResults(body['dist-tags'].latest);
+      } else {
+        badgeData.text[1] = 'invalid';
+        badgeData.colorscheme = 'red';
+
+        sendBadge(format, badgeData);
+      }
+    });
+  }
+
+  function separateAndSetFormat (str = '') {
```

Comment:
This is confusing. Could you pull the format out of the regex match and assign it right up top? That way this can be a pure function / would not need to set `format`. It's the way all the other badges handle it.